src/modules/ModuleWithItemsFromConfig.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+namespace WPML\PB\Cornerstone\Modules;
+
+use WPML\FP\Obj;
+
+class ModuleWithItemsFromConfig extends \WPML_Cornerstone_Module_With_Items {
+
+	/** @var array $fieldDefinitions */
+	private $fieldDefinitions = [];
+
+	public function __construct( array $config ) {
+		foreach ( $config as $fieldConfig ) {
+			$field                            = Obj::prop( 'field', $fieldConfig );
+			$this->fieldDefinitions[ $field ] = $fieldConfig;
+		}
```

Comment:
@strategio You could do:
```php
		$keyByField = Fns::converge( Lst::zipObj(), [ Lst::pluck( 'field' ), Fns::identity() ] );
		$this->fieldDefinitions = $keyByField( $config );
```

It's a pity we don't have a `Lst::keyBy( $key, $array )` function already.

Reply:
Thanks Bruce, added.